File: app/src/main/java/net/tjado/usbgadget/ViewPager2FragmentStateAdapter.java
package net.tjado.usbgadget;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentActivity;
import androidx.viewpager2.adapter.FragmentStateAdapter;

public class ViewPager2FragmentStateAdapter extends FragmentStateAdapter {

    ViewPager2FragmentStateAdapter(@NonNull FragmentActivity fragmentActivity) {
        super(fragmentActivity);
    }

    @NonNull
    @Override
    public Fragment createFragment(int position) {
        switch (position) {
            case 0:
                return new OverviewFragment();
            case 1:
                return new DeviceInfoFragment();
            case 2:
                return new LogFragment();
        }
        return null;
    }

    @Override
    public int getItemCount() {
        return 3;
    }
}

File: app/src/main/java/net/tjado/usbgadget/ViewPager2FragmentStateAdapter.kt
package net.tjado.usbgadget

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentActivity
import androidx.viewpager2.adapter.FragmentStateAdapter

class ViewPager2FragmentStateAdapter internal constructor(fragmentActivity: FragmentActivity) :
    FragmentStateAdapter(fragmentActivity) {
    override fun createFragment(position: Int): Fragment {
        return when (position) {
            1 -> DeviceInfoFragment()
            2 -> GadgetManagerFragment()
            3 -> LogFragment()
            else -> OverviewFragment()
        }
    }

    override fun getItemCount(): Int {
        return 4
    }
}